### DISCUSSION TITLE
Teamspeak Template ARM64

### DISCUSSION THREAD

**Discussion by Vinookieadv** [pufferpanel/pufferpanel#1284] [Q&A]
Hey there, anyone could help me to configure a new template?
I wanted to use this docker for starting:

sudo docker run -d --name TeamSpeak -e TS_UPDATE=1 -e TS_UPDATE_BACKUP=1 -e TIME_ZONE=Europe/Berlin -p 9987:9987/udp -p 10011:10011/tcp -p 30033:30033/tcp -v /opt/teamspeak/:/teamspeak/save/ ertagh/teamspeak3-server:latest 

cat /opt/teamspeak/logs/*

if i use the code below the pufferpanel console ask me for the pufferpanel user password. What is the pufferpanel password, can i just change it? If i leave it blank i get "wrong password".

i tried to use this code as template:

{
  "name": "teamspeak3",
  "display": "Teamspeak 3",
  "type": "voice",
  "install": [
        {
      "commands": [
        "sudo docker run -d --name TeamSpeak -e TS_UPDATE=1 -e TS_UPDATE_BACKUP=1 -e TIME_ZONE=Europe/Berlin -p 9987:9987/udp -p 10011:10011/tcp -p 30033:30033/tcp -v /opt/teamspeak/:/teamspeak/save/ ertagh/teamspeak3-server:latest"
      ],
      "type": "command"
    }
  ],
  "run": {
    "stopCode": 2,
    "command": "cat /opt/teamspeak/logs/*",
    "workingDirectory": "",
    "pre": [],
    "post": [],
    "environmentVars": {}
  },
  "data": {
    "eula": {
      "type": "option",
      "desc": "Do you accept the Teamspeak EULA? 1 = Yes, 0 = No.",
      "display": "EULA",
      "required": true,
      "value": "0",
      "options": [
        {
          "value": "0",
          "display": "No"
        },
        {
          "value": "1",
          "display": "Yes"
        }
      ]
    },
    "ftport": {
      "type": "integer",
      "desc": "What Port do you want your file transfer server to run on?",
      "display": "Port Number",
      "value": "30033"
    },
    "ip": {
      "type": "string",
      "desc": "What IP do you want your voice server to run on?",
      "display": "IP Address",
      "value": "0.0.0.0"
    },
    "license": {
      "type": "string",
      "desc": "The absolute path to your license file.",
      "display": "License Path",
      "value": ""
    },
    "qport": {
      "type": "string",
      "desc": "What Port do you want your query server to run on?",
      "display": "Port Number",
      "value": "10011"
    },
    "version": {
      "type": "string",
      "desc": "What version of Teamspeak do you want to use?",
      "display": "Version",
      "required": true,
      "value": "3.13.6"
    },
    "vport": {
      "type": "integer",
      "desc": "What Port do you want your voice server to run on?",
      "display": "Port Number",
      "value": "9987"
    }
  },
  "environment": {
    "type": "tty"
  },
  "supportedEnvironments": [
    {
      "type": "tty"
    }
  ],
  "requirements": {
    "binaries": [
      "tar"
    ],
    "os": "linux",
    "arch": "arm64"
  }
}

**Comment by nepcore** [comment]
The `pufferpanel` user does not come with a password as it's not meant to be logged into, it also does not have sudo access and neither should it

To run a server in a Docker container you don't use `docker run` directly, you should use the Docker environment built into PufferPanel instead, take a look at a few of the existing Docker variants of templates for examples on how that works

**Comment by Vinookieadv** [comment]
Got so far, but still no success.
maybe you have an other hint for me?:
```
{
  "name": "Teamspeak",
  "display": "Teamspeak3 (Docker)",
  "type": "docker",
  "install": [
    {
      "commands": [
        "sudo docker pull ertagh/teamspeak3-server:latest"
      ],
      "type": "command"
    }
  ],
  "run": {
    "stop": "sudo docker stop TeamSpeak || true",
    "command": "sudo docker run -d --name TeamSpeak -e TS3SERVER_LICENSE=accept -e TS_UPDATE=1 -e TS_UPDATE_BACKUP=1 -e TIME_ZONE=Europe/Berlin -p 9987:9987/udp -p 10011:10011/tcp -p 30033:30033/tcp -v /opt/teamspeak:/teamspeak/save ertagh/teamspeak3-server:latest",
    "workingDirectory": "",
    "pre": [],
    "post": [],
    "environmentVars": {}
  },
  "data": {},
  "environment": {
    "image": "ertagh/teamspeak3-server:latest",
    "type": "docker"
  },
  "supportedEnvironments": [
    {
      "image": "ertagh/teamspeak3-server:latest",
      "type": "docker"
    }
  ],
  "requirements": {
    "arch": "amd64"
  }
}
```
